src/merkle/simple_smt/mod.rs

```diff
@@ -254,3 +248,18 @@ impl SimpleSmt {
         self.branches.insert(index, branch);
     }
 }
+
+// BRANCH NODE
+// ================================================================================================
+
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+struct BranchNode {
```

Comment:
Shouldn't this be deprecated in favor of `MerkleStore`?

Reply:
Do you mean the whole `SimpleSmt` or just the `BranchNode`?